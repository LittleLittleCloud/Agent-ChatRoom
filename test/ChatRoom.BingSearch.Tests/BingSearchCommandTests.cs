using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using Spectre.Console.Testing;
using Xunit;

namespace ChatRoom.BingSearch.Tests;

public class BingSearchCommandTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public async Task ItShowHelperTextTestAsync()
    {
        var app = new CommandAppTester();
        app.SetDefaultCommand<BingSearchCommand>(BingSearchCommand.Description);

        var result = await app.RunAsync("--help");

        Approvals.Verify(result.Output);
    }
}
