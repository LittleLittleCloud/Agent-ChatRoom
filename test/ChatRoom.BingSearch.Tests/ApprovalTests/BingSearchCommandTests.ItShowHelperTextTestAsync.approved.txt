DESCRIPTION:
Bing search agent for chat room.
The agent will search the query from Bing search engine and return the result.

To use the agent, you need to provide a configuration file.
A configuration file is a json file with the following schema:
- https://raw.githubusercontent.com/LittleLittleCloud/Agent-ChatRoom/main/schema/chatroom_bing_search_configuration_schema.json

USAGE:
    testhost.dll [OPTIONS]

OPTIONS:
    -h, --help               Prints help information
    -v, --version            Prints version information
    -c, --config <CONFIG>    Configuration file